crates/signer/src/error.rs

```diff
@@ -79,7 +79,12 @@ pub enum UnsupportedSignerOperation {
 
 impl fmt::Display for UnsupportedSignerOperation {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.as_str().fmt(f)
+        match self {
```

Comment:
you want to change this above in enum Error, but i think it should be another variant

Reply:
do you mean convert UnsupportedSignerOperation into an error via thiserror?

I think we should do this because otherwise this new message will mess with the error message here:

https://github.com/alloy-rs/alloy/blob/750004c1ce56b5044b27d19a980cb3b7d1efc898/crates/signer/src/error.rs#L13-L14

Reply:
I think we should keep things as is and just add another variant with a message that it is intentionally unimplemented

Reply:
right. Your suggestion is better by making a distinction between what is unsupported and what is opted out due to security reasons. maybe a new variant to the Error definition:
```
pub enum Error {
    // ...
    
    /// The operation is deliberately disabled for security reasons
    #[error("Operation {0} is restricted due to security restrictions")]
    SecurityRestriction(String),
    
    // ...
}
```